The job **`dist-aarch64-msvc`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/6688897066/job/18171615015) [(plain)](https://github.com/rust-lang-ci/rust/commit/db832cde347c3b3ab792c9a66736557bebdb5d5c/checks/18171615015/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
   Compiling clap v4.4.7
   Compiling clap_complete v4.4.3
   Compiling build_helper v0.1.0 (C:\a\rust\rust\src\tools\build_helper)
   Compiling xz2 v0.1.6
error[E0432]: unresolved import `windows::Win32::System::ProcessStatus::K32GetProcessMemoryInfo`
    |
280 |             K32GetProcessMemoryInfo, PROCESS_MEMORY_COUNTERS, PROCESS_MEMORY_COUNTERS_EX,
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             |
    |             no `K32GetProcessMemoryInfo` in `Windows::Win32::System::ProcessStatus`
    |             help: a similar name exists in the module: `GetProcessMemoryInfo`
For more information about this error, try `rustc --explain E0432`.
error: could not compile `bootstrap` (bin "rustc") due to previous error
warning: build failed, waiting for other jobs to finish...
failed to run: C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin\cargo.exe build --manifest-path C:\a\rust\rust\src/bootstrap/Cargo.toml --locked --features build-metrics
failed to run: C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin\cargo.exe build --manifest-path C:\a\rust\rust\src/bootstrap/Cargo.toml --locked --features build-metrics
Build completed unsuccessfully in 0:00:44
make: *** [Makefile:65: prepare] Error 1
Command failed. Attempt 2/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0432]: unresolved import `windows::Win32::System::ProcessStatus::K32GetProcessMemoryInfo`
    |
280 |             K32GetProcessMemoryInfo, PROCESS_MEMORY_COUNTERS, PROCESS_MEMORY_COUNTERS_EX,
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             |
    |             no `K32GetProcessMemoryInfo` in `Windows::Win32::System::ProcessStatus`
    |             help: a similar name exists in the module: `GetProcessMemoryInfo`
For more information about this error, try `rustc --explain E0432`.
error: could not compile `bootstrap` (bin "rustc") due to previous error
failed to run: C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin\cargo.exe build --manifest-path C:\a\rust\rust\src/bootstrap/Cargo.toml --locked --features build-metrics
Build completed unsuccessfully in 0:00:00
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:65: prepare] Error 1
Command failed. Attempt 3/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0432]: unresolved import `windows::Win32::System::ProcessStatus::K32GetProcessMemoryInfo`
    |
280 |             K32GetProcessMemoryInfo, PROCESS_MEMORY_COUNTERS, PROCESS_MEMORY_COUNTERS_EX,
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             |
    |             no `K32GetProcessMemoryInfo` in `Windows::Win32::System::ProcessStatus`
    |             help: a similar name exists in the module: `GetProcessMemoryInfo`
For more information about this error, try `rustc --explain E0432`.
error: could not compile `bootstrap` (bin "rustc") due to previous error
failed to run: C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin\cargo.exe build --manifest-path C:\a\rust\rust\src/bootstrap/Cargo.toml --locked --features build-metrics
Build completed unsuccessfully in 0:00:00
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:65: prepare] Error 1
Command failed. Attempt 4/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0432]: unresolved import `windows::Win32::System::ProcessStatus::K32GetProcessMemoryInfo`
    |
280 |             K32GetProcessMemoryInfo, PROCESS_MEMORY_COUNTERS, PROCESS_MEMORY_COUNTERS_EX,
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             |
    |             no `K32GetProcessMemoryInfo` in `Windows::Win32::System::ProcessStatus`
    |             help: a similar name exists in the module: `GetProcessMemoryInfo`
For more information about this error, try `rustc --explain E0432`.
error: could not compile `bootstrap` (bin "rustc") due to previous error
failed to run: C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin\cargo.exe build --manifest-path C:\a\rust\rust\src/bootstrap/Cargo.toml --locked --features build-metrics
Build completed unsuccessfully in 0:00:00
Build completed unsuccessfully in 0:00:00
make: *** [Makefile:65: prepare] Error 1
Command failed. Attempt 5/5:
##[group]Building bootstrap
   Compiling bootstrap v0.0.0 (C:\a\rust\rust\src\bootstrap)
error[E0432]: unresolved import `windows::Win32::System::ProcessStatus::K32GetProcessMemoryInfo`
    |
280 |             K32GetProcessMemoryInfo, PROCESS_MEMORY_COUNTERS, PROCESS_MEMORY_COUNTERS_EX,
    |             ^^^^^^^^^^^^^^^^^^^^^^^
    |             |
    |             |
    |             no `K32GetProcessMemoryInfo` in `Windows::Win32::System::ProcessStatus`
    |             help: a similar name exists in the module: `GetProcessMemoryInfo`
For more information about this error, try `rustc --explain E0432`.
error: could not compile `bootstrap` (bin "rustc") due to previous error
failed to run: C:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin\cargo.exe build --manifest-path C:\a\rust\rust\src/bootstrap/Cargo.toml --locked --features build-metrics
Build completed unsuccessfully in 0:00:00
```

</details>

--- a/src/bootstrap/Cargo.toml
+++ b/src/bootstrap/Cargo.toml
@@ -35,7 +35,7 @@ test = false
 [dependencies]
 build_helper = { path = "../tools/build_helper" }
 cc = "1.0.69"
-clap = { version = "4.2.4", default-features = false, features = ["std", "usage", "help", "derive", "error-context"] }
+clap = { version = "4.4.7", default-features = false, features = ["std", "usage", "help", "derive", "error-context"] }
 clap_complete = "4.4.3"
 cmake = "0.1.38"
 filetime = "0.2"
@@ -64,13 +64,13 @@ sysinfo = { version = "0.26.0", optional = true }
 
 # Solaris doesn't support flock() and thus fd-lock is not option now
 [target.'cfg(not(target_os = "solaris"))'.dependencies]
-fd-lock = "3.0.8"
+fd-lock = "3.0.13"
 
 [target.'cfg(windows)'.dependencies.junction]
 version = "1.0.0"
 
 [target.'cfg(windows)'.dependencies.windows]
-version = "0.46.0"
+version = "0.51.1"
 features = [
     "Win32_Foundation",
     "Win32_Security",

--- a/src/bootstrap/src/bin/rustc.rs
+++ b/src/bootstrap/src/bin/rustc.rs
@@ -300,10 +300,9 @@ fn format_rusage_data(child: Child) -> Option<String> {
             &mut user_filetime,
         )
     }
-    .ok()
     .ok()?;
-    unsafe { FileTimeToSystemTime(&user_filetime, &mut user_time) }.ok().ok()?;
-    unsafe { FileTimeToSystemTime(&kernel_filetime, &mut kernel_time) }.ok().ok()?;
+    unsafe { FileTimeToSystemTime(&user_filetime, &mut user_time) }.ok()?;
+    unsafe { FileTimeToSystemTime(&kernel_filetime, &mut kernel_time) }.ok()?;
 
     // Unlike on Linux with RUSAGE_CHILDREN, this will only return memory information for the process
     // with the given handle and none of that process's children.

--- a/src/bootstrap/src/utils/job.rs
+++ b/src/bootstrap/src/utils/job.rs
@@ -89,8 +89,7 @@ mod for_windows {
             JobObjectExtendedLimitInformation,
             &info as *const _ as *const c_void,
             mem::size_of_val(&info) as u32,
-        )
-        .ok();
+        );
         assert!(r.is_ok(), "{}", io::Error::last_os_error());
 
         // Assign our process to this job object. Note that if this fails, one very
@@ -102,9 +101,9 @@ mod for_windows {
         // Also note that nested jobs (why this might fail) are supported in recent
         // versions of Windows, but the version of Windows that our bots are running
         // at least don't support nested job objects.
-        let r = AssignProcessToJobObject(job, GetCurrentProcess()).ok();
+        let r = AssignProcessToJobObject(job, GetCurrentProcess());
         if r.is_err() {
-            CloseHandle(job);
+            CloseHandle(job).ok();
             return;
         }
 
@@ -131,7 +130,7 @@ mod for_windows {
                 // it might be better to improve the experience of the second case
                 // when users have interrupted the parent process and we haven't finish
                 // duplicating the handle yet. We just need close the job object if that occurs.
-                CloseHandle(job);
+                CloseHandle(job).ok();
                 return;
             }
         };
@@ -145,8 +144,7 @@ mod for_windows {
             0,
             false,
             DUPLICATE_SAME_ACCESS,
-        )
-        .ok();
+        );
 
         // If this failed, well at least we tried! An example of DuplicateHandle
         // failing in the past has been when the wrong python2 package spawned this
@@ -155,7 +153,7 @@ mod for_windows {
         // mode" here is that we only clean everything up when the build system
         // dies, not when the python parent does, so not too bad.
         if r.is_err() {
-            CloseHandle(job);
+            CloseHandle(job).ok();
         }
     }
 }

